packages/web/src/common/store/cache/tracks/utils/retrieveTracks.ts

```diff
@@ -84,9 +91,12 @@ export function* retrieveTrackByHandleAndSlug({
       },
       kind: Kind.TRACKS,
       idField: 'track_id',
-      forceRetrieveFromSource,
+      // If this is the first fetch after server side rendering the track page,
+      // force retrieve from source to ensure we have personalized data
+      forceRetrieveFromSource:
+        forceRetrieveFromSource ?? isInitialFetchAfterSsr,
       shouldSetLoading: true,
-      deleteExistingEntry: false,
+      deleteExistingEntry: isInitialFetchAfterSsr,
```

Comment:
cool so the move is to all all items from ssr to be replaceable?

Reply:
Yes, in the future it would be ideal to use trpc to fetch only the personalized data. But for now it's easiest to fetch the whole object and clobber the existing cache entry